src/App.tsx

```diff
@@ -1,39 +1,44 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
-function getRandomName(): string {
-  const value = Date.now().toString().slice(-4);
+type State = {
+  isContextmenu: boolean,
+};
 
-  return `Clock-${value}`;
-}
+export class App extends Component<{}, State> {
+  state = {
+    isContextmenu: true,
+  };
 
-export const App: React.FC = () => {
-  const today = new Date();
-  let clockName = 'Clock-0';
+  componentDidMount() {
+    document.addEventListener('contextmenu', this.handleContextMenu);
+    document.addEventListener('click', this.handleClick);
+  }
 
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    clockName = getRandomName();
-  }, 3300);
+  componentWillUnmount() {
+    document.removeEventListener('contextmenu', this.handleContextMenu);
+    document.removeEventListener('click', this.handleClick);
+  }
 
-  // this code stops the timer
-  window.clearInterval(timerId);
+  handleContextMenu = (e: MouseEvent) => {
+    e.preventDefault();
+    this.setState({ isContextmenu: false });
+  };
 
-  return (
-    <div className="App">
-      <h1>React clock</h1>
+  handleClick = () => {
+    this.setState({ isContextmenu: true });
+  };
 
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
+  render() {
+    const { isContextmenu } = this.state;
 
-        {' time is '}
+    return (
+      <div className="App">
+        <h1>React clock</h1>
 
-        <span className="Clock__time">
-          {today.toUTCString().slice(-12, -4)}
-        </span>
+        {isContextmenu && <Clock />}
```

Comment:
Clock should receive the name
![image](https://user-images.githubusercontent.com/98764641/237044742-e9d1d2b4-ff14-40be-a136-315f845890c1.png)